src/main/java/net/minecraftforge/server/permission/PermissionAPI.java

```diff
@@ -19,80 +19,73 @@
-     * @param desc  Optional description of the node
-     */
-    public static String registerNode(String node, DefaultPermissionLevel level, String desc)
-    {
-        Preconditions.checkNotNull(node, "Permission node can't be null!");
-        Preconditions.checkNotNull(level, "Permission level can't be null!");
-        Preconditions.checkNotNull(desc, "Permission description can't be null!");
-        Preconditions.checkArgument(!node.isEmpty(), "Permission node can't be empty!");
-        // TODO Loader states Preconditions.checkState(Loader.instance().getLoaderState().ordinal() > LoaderState.PREINITIALIZATION.ordinal(), "Can't register permission nodes before Init!");
-        permissionHandler.registerNode(node, level, desc);
-        return node;
-    }
+   /**
+    * <p>Queries a player's permission for a given node and contexts</p>
+    * <p><strong>Warning:</strong> PermissionNodes <strong>must</strong> be registered using the
+    * {@link PermissionGatherEvent.Nodes} event before querying.</p>
+    *
+    * @param player player for which you want to check permissions
+    * @param node the PermissionNode for which you want to query
+    * @param context optional array of PermissionDynamicContext, single entries will be ignored if they weren't
+    *                registered to the node
+    * @param <T></T> type of the queried PermissionNode
+    * @return a value of type {@code <T>}, that the combination of Player and PermissionNode map to, defaults to the
+    * PermissionNodes default handler.
+    */
+   public static <T> T getPermissions(ServerPlayer player, PermissionNode<T> node, PermissionDynamicContext<? extends StringRepresentable>... context)
+   {
+      return activeHandler.getPermissions(player, node, context);
+   }
 
-    /**
-     * @param profile GameProfile of the player who is requesting permission. The player doesn't have to be online
-     * @param node    Permission node. See {@link #registerNode(String, DefaultPermissionLevel, String)}
-     * @param context Context for this permission. Highly recommended to not be null. See {@link IContext}
-     * @return true, if player has permission, false if he does not.
-     * @see DefaultPermissionHandler
-     */
-    public static boolean hasPermission(GameProfile profile, String node, @Nullable IContext context)
-    {
-        Preconditions.checkNotNull(profile, "GameProfile can't be null!");
-        Preconditions.checkNotNull(node, "Permission node can't be null!");
-        Preconditions.checkArgument(!node.isEmpty(), "Permission node can't be empty!");
-        return permissionHandler.hasPermission(profile, node, context);
-    }
+   /**
+    * <p>Helper method for internal use only!</p>
+    * <p>Fires the {@link PermissionGatherEvent.Handler}  event,
+    * and replaced the current PermissionHandler with the one returned by the event.</p>
+    */
+   public static void gatherPermissionHandler()
+   {
```

Comment:
Generally check your formatting.
Won't continue reviewing the formatting for the remaining files.